### ISSUE TITLE
Maxpooling issue

### ISSUE THREAD

**Issue by uu-praveeng** [01org/mkl-dnn#45]
Hi,

I am trying to use the maxpooling for MNIST digit classification.
I am using the reorder to convert ,input from nchw to nChw8c format and then doing the Maxpooling considering output to be in nChw8c  format , and then doing output reorder from nChw8c to nchw.

Here I am unable to get the desired output for below dimensions 
for input[1,20,25,25] , max pool 2,2 :stride 2,2 pad 0,0 

However if I  consider nchw format for input/ouput without reorder I am getting the desired result.
Please suggest if I am missing anything.

Thanks,
G.Praveen.

**Comment by emfomenk** [comment]
Hi @uu-praveeng,

At this moment blocked formats (such as `nChw8c`) assume C (channels) to be multiple of corresponding value (for `nChw8c` we block channels into blocks of size 8). Your input has 20 channels, which is not multiple of 8, hence it is not possible to use this format for the case.

In future we probably will allow using blocked format even in cases when corresponding dimension is not divided by block size by rounding up former, i.e. considering 20 channels as 24, but this is not implemented yet.

Long story short -- please use non-blocked format for this particular case.

**Comment by uu-praveeng** [comment]
Thank you.
